### DISCUSSION TITLE
error[E0463]: can't find crate for `std` out of nowhere

### DISCUSSION THREAD

**Discussion by ethindp** [phil-opp/blog_os#815] [Questions]
So I've done some reorganization of my kernel source tree, and moved some things into their own crates. However, now rust is telling me it can't find the std crate even though I've explicitly marked all the crates I've made as no_std and have attempted to ensure that all the deps I pull in for those crates are no_std as well. I've pulled in crates like crc, heapless, itertools, anyhow, zerocopy, uuid, and static_assertions, but so far as I know none of those crates depend on the standard library, or I've disabled the default features to ensure they don't. So I'm at a loss as to why it suddenly doesn't want to build and demands stdlib support. Any ideas?
https://github.com/ethindp/kernel is the repository.

**Comment by phil-opp** [comment]
I just tried to compile your kernel and I got the following error:

```
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-kernel-none-7873401560910942873` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `crc`.
```

According to the last line, the problem seems to be that the `crc` crate depends on libstd. To fix this, you have to disable the default features for the `crc` dependency in drivers/storage/part/Cargo.toml.

**Comment by ethindp** [comment]
Yeah, I just wrote my own CRC algorithm in the crate itself as a
private function. I might want to contribute my various driver crates
to the rust-osdev project -- I believe they work, though I'll need to
(maybe) document them a bit more and write test suites for them. I've
only got the part crate for parsing partition tables (GPT only for
now). Though both MBR and GPT crates already exist, they are for
libstd environments, not no_std ones, so I wrote my own. I'd be happy
to outline how it works in another issue, or even on the Gitter
channel, and not here because I'd be dragging this issue off-topic.

On 6/21/20, Philipp Oppermann <notifications@github.com> wrote:
> I just tried to compile your kernel and I got the following error:
>
> ```
> error[E0463]: can't find crate for `std`
>   |
>   = note: the `x86_64-kernel-none-7873401560910942873` target may not be
> installed
>
> error: aborting due to previous error
>
> For more information about this error, try `rustc --explain E0463`.
> error: could not compile `crc`.
> ```
>
> According to the last line, the problem seems to be that the `crc` crate
> depends on libstd. To fix this, you have to disable the default features for
> the `crc` dependency in drivers/storage/part/Cargo.toml.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/phil-opp/blog_os/issues/815#issuecomment-647124008


-- 
Signed,
Ethin D. Probst

**Comment by ergpopler** [comment]
@ethindp you've gotten it to work? I am having the same issue.
